src/loader.js

```diff
@@ -0,0 +1,91 @@
+let assume = require('assume');
+let debug = require('debug')('taskcluster:loader');
+let Promise = require('promise');
+
+function loader(componentDirectory, requiredVirtualComponents) {
+  /**
+   * If present, this is a list of component names that a loader
+   * must define.  These are likely things like configuration values
+   */
+  if (requiredVirtualComponents) {
+    assume(requiredVirtualComponents).is.array();
+  } else {
+    requiredVirtualComponents = [];
+  }
```

Comment:
I'm happy to build a graphviz target, but I think that I think that forcing all virtual dependencies to be flat and all componentDirectory ones to be the object style isn't the right thing to do, which is what's happening right now in that code... I'll take a look at it and adapt it.